### PR TITLE
ar71xx: add support for Yuncore XD6700

### PR DESCRIPTION
**Pull request by ruairi-hickey**
Yuncore XD6700 is a router based on Qualcomm Atheros QCA9533 ver 2 rev 0

Specification:

650/597/216 MHz (CPU/DDR/AHB)
128 MB of RAM (DDR)
16 MB of FLASH (SPI NOR)
Ath9K 2.4 GHz
Ath10K 5.8GHZ (QCA9887)
2 x 10/100/1000 Mbps Ethernet 24V Passive PoE (AG71XX)
1 x System LED, 1 x LAN / 1 x WAN LED
1 x button
UART header on PCB
Custom attitude_adjustment/12.09-rc1

Currently not working:

Update from oficial web interface

Flash instruction:

Download lede-ar71xx-generic-xd6700-squashfs-sysupgrade.bin
Set PC to fixed ip address 192.168.1.xx and connect to WAN interface.

Power On Router and during boot the LAN/WAN LED's will blink
After about 15 seconds both LEDS will turn off
Press the Reset button to enter failsafe mode.
(You can Press the Reset Button mutiple times to ensure you enter failsafe mode as there is a small window)

The router will be on ip address 192.168.1.1 and has a telnet daemon listening.

Telnet to the router
run mount_root to remount the root volume as r/w to allow for the (automatic) creation of /dev/mtd1

use scp or otherwise to copy lede-ar71xx-generic-xd6700-squashfs-sysupgrade.bin to /tmp
cd /tmp
mtd -e firmware -r write lede-ar71xx-generic-xd6700-squashfs-sysupgrade.bin firmware

### CODE REVIEW

**Comment by ruairi-hickey** [comment]
Hi,    I have created this patch to add support for the yuncore xd6700 routers.

Thanks,
       Ruairi

**Comment by pepe2k** [comment]
Hello @ruairi-hickey,

Please, read our [submitting patches rules](https://lede-project.org/submitting-patches), and make sure your pull request follows them. Your commit is missing a proper prefix and subject. Please, include in your commit message also a short description of the hardware and how to install LEDE on it. Have a look at the [recent additions for some examples](https://git.lede-project.org/?p=source.git&a=search&h=HEAD&st=commit&s=add+support+for).

Cheers,
Piotr

File: target/linux/ar71xx/files/arch/mips/ath79/Kconfig.openwrt [hunk 1]
```diff
@@ -482,12 +482,24 @@ config ATH79_MACH_WPJ563
...
+        select ATH79_DEV_ETH
+        select ATH79_DEV_GPIO_BUTTONS
+        select ATH79_DEV_LEDS_GPIO
+        select ATH79_DEV_M25P80
+        select ATH79_DEV_USB
+        select ATH79_DEV_WMAC
+
+config ATH79_MACH_XD6700
+	bool "YunCore XD6700 support"
+        select SOC_QCA956X
```
Comment (hauke) [inline target/linux/ar71xx/files/arch/mips/ath79/Kconfig.openwrt]: It looks like the indention is wrong here and also in the parts on top of this

**Comment by ruairi-hickey** [comment]
OK, I'll get to fixing this some evening this week, and will update the request.

Ruairi

**Comment by ruairi-hickey** [comment]
Hi @pepe2k ,
    I corrected the indentation / comment etc however I am very new to git etc, this is the 1st time I have used it so I may not have updated this correctly.  Please let me know if I need to make more changes, submit the changes is a different way.

Thanks,
      Ruairi

File: target/linux/ar71xx/files/arch/mips/ath79/Kconfig.openwrt (hunk 1 above)
Comment (ruairi-hickey) [reply target/linux/ar71xx/files/arch/mips/ath79/Kconfig.openwrt]: Hi @hauke ,
  I believe I have corrected all the indentation errors etc.  Let me know if I have not as I am unfamiliar with git, submitting to projects.

Thanks,
        Ruairi

**Comment by ruairi-hickey** [comment]
Do I need to do something to resubmit this ?

Thanks,
          Ruairi

**Comment by pepe2k** [comment]
@ruairi-hickey, your PR has 17 commits, please squash them into single one.

Cheers,
Piotr

**Comment by ruairi-hickey** [comment]
Hi Piotr,
         I have squashed the patch into a single merge.  I hope this is OK now...

Thanks for all the help,

Ruairi

**Comment by ruairi-hickey** [comment]
Hi Piotr,
        Could you let me know if the single commit is OK or if I need to make any further changes.

Thanks
         Ruairi

**Comment by blogic** [comment]
With the remerge in progress, all PRs on the lede-project organisation will be closed. Please help getting this merged or rebase/post it on the openwrt project page (https://github.com/openwrt/openwrt/pulls). All remaining PRs will be closed in 30 days.

**Comment by ruairi-hickey** [comment]
Hi,
    I have created a pull request on the openwrt project page.  Should I close this request ?

Thanks,
         Ruairi

**Comment by pepe2k** [comment]
Superseded by https://github.com/openwrt/openwrt/pull/612.

**Pull request closed without merging**